### PR TITLE
[deprecation] Remove fix in favor of set_value

### PR DESCRIPTION
**Pull request by odow**
JuMP's Nonlinear parameters use `set_value` and `value`: https://jump.dev/JuMP.jl/stable/nlp/#Nonlinear-Parameters

We should follow that.

### CODE REVIEW

File: src/variable_interface.jl [hunk 1]
```diff
@@ -115,44 +74,3 @@ function parameter_by_name(model::Model, name::String)
     end
     return nothing
 end
-
-JuMP.has_lower_bound(p::ParameterRef) = false
```
Comment (joaquimg) [inline src/variable_interface.jl:119]: Are the errors for missing methods very clear about what the user should do?

**Review by joaquimg — COMMENTED** [review]
Set value is ok!
Not sure about the rest.

File: src/variable_interface.jl (hunk 1 above)
Comment (odow) [reply src/variable_interface.jl:119]: It will throw a method error. This is not defined.

Comment (odow) [reply src/variable_interface.jl:119]: We don't define `has_lower_bound` on nonlinear parameters, for example.

**Comment by odow** [comment]
Looks like my merge didn't work. Let me try again.

File: src/variable_interface.jl (hunk 1 above)
Comment (odow) [reply src/variable_interface.jl:119]:
```Julia
julia> has_lower_bound(p)
ERROR: MethodError: no method matching has_lower_bound(::ParameterRef)
Closest candidates are:
  has_lower_bound(::VariableRef) at /Users/oscar/.julia/packages/JuMP/y5vgk/src/variables.jl:352
Stacktrace:
 [1] top-level scope at REPL[9]:1
```

Comment (joaquimg) [reply src/variable_interface.jl:119]: I see, well ok. Lets got for it and think more if people get confused at some point.
